src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -1546,7 +1546,7 @@ public override IMethodSymbol GetDeclaredSymbol(ArrowExpressionClauseSyntax decl
                     // We are looking for the SourcePropertyAccessorSymbol here,
                     // not the SourcePropertySymbol, so use declarationSyntax
                     // to exclude the property symbol from being retrieved.
-                    return this.GetDeclaredMember(container, declarationSyntax.Span) as MethodSymbol;
+                    return (this.GetDeclaredMember(container, declarationSyntax.Span) as MethodSymbol).GetPublicSymbol();
 
```

Comment:
Does this (and similar patterns) need a `?.GetPublicSymbol()` in case GetDeclaredMember is returning null? It seems like this was previously being defensive and the defense is now lost?

Reply:
> Does this (and similar patterns) need a ?.GetPublicSymbol() in case GetDeclaredMember is returning null? It seems like this was previously being defensive and the defense is now lost?

```GetPublicSymbol``` is an extension method that is safe to call with a null receiver. It does the null check itself.